### PR TITLE
IssueFix1196

### LINKED ISSUE #1196: Should AppConfig.sBackgroundHandlerThread still be static?
OK, this is not the kind of programming style issue like @EmmanuelMess raised from time to time, but is more practical issue on writing tests (related to #959).

On writing test case for #1195 I couldn't run more than one test case at a time.

```
java.lang.IllegalThreadStateException
	at java.lang.Thread.start(Thread.java:705)
	at com.amaze.filemanager.utils.application.AppConfig.onCreate(AppConfig.java:76)
	at org.robolectric.android.internal.ParallelUniverse.lambda$setUpApplicationState$0(ParallelUniverse.java:173)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:72)
	at org.robolectric.android.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:172)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:328)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:241)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:123)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:77)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)
```

Turned out the culprit was `AppConfig.sBackgroundHandlerThread`

https://github.com/TeamAmaze/AmazeFileManager/blob/e8da864ecf5216f5354eb0f1b3e5bf73c610ad86/app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java#L37

which survived between test cases, but as the thread was terminated after the first test case was run and `AppConfig.onTerminate()` was called, the second test case would certainly fail when `AppConfig.onCreate()` attempted to start this terminated thread.

Hence my question: should `AppConfig.sBackgroundHandlerThread` still be static?

### PR DESCRIPTION
**Pull request by beastlyCoder**
I was able to fix the issue listed in issue 1196. I removed the static field and initialized it inside of the method. It is still an instance variable, but instead it is only declared and not initialized.

Fix #1196

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java [hunk 1]
```diff
@@ -51,7 +51,10 @@
     private UtilsHandler mUtilsHandler;
 
     private static Handler mApplicationHandler = new Handler();
-    private static HandlerThread sBackgroundHandlerThread = new HandlerThread("app_background");
+    //This caused the IllegalThreadStateException because it is being declared and initialized
+    //at the same time, as @TranceLove said on stack overflow issue 1196
+    //private static HandlerThread sBackgroundHandlerThread = new HandlerThread("app_background");
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java]: Why is this line here?

File: app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java [hunk 2]
```diff
@@ -51,7 +51,10 @@
     private UtilsHandler mUtilsHandler;
 
     private static Handler mApplicationHandler = new Handler();
-    private static HandlerThread sBackgroundHandlerThread = new HandlerThread("app_background");
+    //This caused the IllegalThreadStateException because it is being declared and initialized
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java]: This comment is too specific to an issue in github, only use these type of comments for hacks. If needed, use javadoc explaining the variable.

Comment (beastlyCoder) [reply app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java]: Oh so for future reference comments like this are not good, is it better for me to make these sorts of descriptions in my pull request?

**Comment by beastlyCoder** [comment]
@EmmanuelMess Is the newest commit acceptable?

File: app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java (hunk 2 above)
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/utils/application/AppConfig.java]: Surely

**Comment by EmmanuelMess** [comment]
I would think so, @TranceLove ?

**Comment by TranceLove** [comment]
Sorry for replying late. Did think of fixing this but on a different scale... Never mind, and that approach didn't succeed yet.

I think it should go just fine but please allow me to run the routine test after sunrise.

**Pull request merged**